manager/api/rest-impl/src/main/java/io/apiman/manager/api/rest/impl/PermissionsResourceImpl.java

```diff
@@ -1,99 +0,0 @@
-/*
- * Copyright 2014 JBoss Inc
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package io.apiman.manager.api.rest.impl;
-
-import io.apiman.manager.api.beans.idm.UserPermissionsBean;
-import io.apiman.manager.api.core.IStorageQuery;
-import io.apiman.manager.api.core.exceptions.StorageException;
-import io.apiman.manager.api.rest.contract.IPermissionsResource;
-import io.apiman.manager.api.rest.contract.exceptions.NotAuthorizedException;
-import io.apiman.manager.api.rest.contract.exceptions.SystemErrorException;
-import io.apiman.manager.api.rest.contract.exceptions.UserNotFoundException;
-import io.apiman.manager.api.rest.impl.util.ExceptionFactory;
-import io.apiman.manager.api.security.ISecurityContext;
-
-import javax.enterprise.context.ApplicationScoped;
-import javax.inject.Inject;
-
-/**
- * Implementation of the Permissions API.
- * 
- * @author eric.wittmann@redhat.com
- */
-@ApplicationScoped
-public class PermissionsResourceImpl implements IPermissionsResource {
```

Comment:
Sorry but I missed how this can be removed... (this diff is a bit large) :)  Is this API no longer needed, or have these methods been moved to another resource impl?

Reply:
No problem, I know this is a big PR, maybe I had to split it up into separate ones :)

I removed all operations for the current user because they are mostly duplicate resources like:

* http://localhost:8080/apiman/permissions/ 
  * This endpoint returns all of the permissions assigned to the currently authenticated user.
* http://localhost:8080/apiman/permissions/userId/
  * This endpoint returns all of the permissions assigned to a specific user.

So I also removed the /currentuser endpoint.
Then I changed the order a little bit. The permission are now in the user api directly.

* http://localhost:8080/apiman/users/userId/permissions

Reply:
OK thank you I understand.  So just to be clear, the *logical* consequence of this change  (from an API perspective) is that "currentuser" is now sort of a reserved userId.  Not really, but in the framework of the REST API it is.  So for example, if my username is "ewittmann" then these are all valid REST endpoints:

`/users/ewittmann`
`/users/ewittmann/organizations`
`/users/ewittmann/apis`
`/users/currentuser/info`

And then all the other currentuser stuff from before is gone and instead you have to route the requests for that stuff using the currently authenticated user's actual **userId**.

I'm cool with all of that except the new `/users/currentuser/info` endpoint, since that's a little bit weird.  Not weird enough for me to not merge the changes, but still - just a little weird. :)

On balance, I think the changes to the API are very good - and I'll live with the tiny bit of weirdness!

Reply:
Yes, all of these are valid endpoints.
Everything of the currentuser/* is moved to /user/{userId}/*.
But only you are allowed to access your stuff.

So if I would call `/users/ewittmann/organizations` this will lead to a forbidden status code. Only an admin is allowed to view all other users stuff.

And yeah `/users/currentuser/info` is a workaround but we have had no better idea at this moment. 

I try to explain it and maybe you have a better idea :)

The problem with these changes is that the UI has to know the username for most of the requests now. Like the whole "My Stuff" Pages (/users/fvolk/apis).

But the UI used `currentuser/info` to get the username to assign the current user to the pages.
So we moved this to `users/currentuser/info` so the UI still gets a username and then we can put this username in all new requests. Otherwise, we are not able to find out the username without more changes in the UI.

Reply:
Yeah - I totally understand everything you've explained (including the workaround).  And I also have no better suggestion.  :)  So let's stick with what you have.